lightkurve/correctors.py

```diff
@@ -487,3 +491,212 @@ def breakpoints(self, campaign):
+        containing first and second order fractional pixel fluxes.
+
+        To solve for the PLD model, we need to minimize the difference squared
+
+        .. math::
+
+            \chi^2 = \sum_i \frac{(y_i - m_i)^2}{\sigma_i^2},
+
+        where $y_i$ is the observed flux value at time $t_i$, by solving
+
+        .. math::
+
+            \frac{\partial \chi^2}{\partial a_l} = 0.
+
+        Parameters
+        ----------
+        aperture_mask : array-like, 'pipeline', 'all', 'threshold', or None
+            A boolean array describing the aperture such that `True` means
+            that the pixel will be used.
+            If None or 'all' are passed, all pixels will be used.
+            If 'pipeline' is passed, the mask suggested by the official pipeline
+            will be returned.
+            If 'threshold' is passed, all pixels brighter than 3-sigma above
+            the median flux will be used.
+        transit_mask : array-like
+            An array of indices to be included in de-trending model. This should
+            exclude cadences which occur during transit.
+        n_components_first : int
+            Number of first-order PLD components to reduce to with PCA.
+        n_components_second : int
+            Number of second-order PLD components to reduce to with PCA.
+
+        Returns
+        -------
+        corrected_lightcurve : :class:`LightCurve` object
+            Returns a corrected lightcurve object. Depending on the input, the
+            returned object will be a `KeplerLightCurve`, `TessLightCurve`, or
+            general `LightCurve` object.
+        """
+
+        # parse aperture
+        aperture = self.tpf._parse_aperture_mask(aperture_mask)
+
+        # find pixel bounds of aperture on tpf
+        xmin, xmax = min(np.where(aperture)[0]),  max(np.where(aperture)[0])
+        ymin, ymax = min(np.where(aperture)[1]),  max(np.where(aperture)[1])
+
+        # crop data cube to include only desired pixels
+        # this is required for superstamps to ensure matrix is invertable
+        flux_crop = self.flux[:, xmin:xmax+2, ymin:ymax+2]
+        flux_err_crop = self.flux_err[:, xmin:xmax+2, ymin:ymax+2]
+        aperture_mask = aperture[xmin:xmax+2, ymin:ymax+2]
+
+        # set transit mask
+        if len(transit_mask) == 0:
+            transit_mask = np.where(self.time)
+        M = lambda x: x[transit_mask]
+
+        #  generate flux light curve from desired pixels
+        lc = M(self.tpf).to_lightcurve(aperture_mask=aperture_mask)
+
+        # set aperture values
+        aperture_vals = np.zeros(aperture_mask.shape)
+        aperture_vals[np.where(aperture_mask)] = 1
+
+        # compute raw flux light curve
+        self.aperture_flux = np.array([f*aperture_vals for f in flux_crop]).reshape(len(flux_crop), -1)
+        rawflux = np.sum(self.aperture_flux.reshape(len(self.aperture_flux), -1), axis=1)
+
+        # first order PLD
+        f1 = self.aperture_flux / rawflux.reshape(-1, 1)
```

Comment:
I find these reshapes a little confusing, and I find `np.atleast_Xd` to be a little more explicit, but this is a personal choice.